.github/workflows/main.yaml

```diff
@@ -0,0 +1,38 @@
+name: CI Testing
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    # Strategy sets up a matrix of python versions to run the tests against
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.5, 3.6, 3.7, 3.8]
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
```

Comment:
you have moved the dev dependancies out of requirements.txt and into requirements-dev.txt, but then only appear to use requirements.txt here.

is that intentional as you are installing pytest etc manully in the test block?

if not:
Would it not be to install the dev dependancies and reference the normal requirements file in there?
might save repetition in setting up pytest etc?

Reply:
More of a holistic approach for the project. I believe a standard `requirements.txt` should be the minimum set of packages necessary for a production-ready installation, at the highest versions the package can support. It's not the cleanest solution when also dealing with how the package gets installed via `setup.py`, but I'd like the two to remain consistent.

Further, I did not think it wise to continue having extra development dependencies tucked into the main requirements file. Separating those requirements into different files makes their use more obvious (although I did miss an opportunity to have `requirements-dev.txt` have a line for `-r requirements.txt` in it, but the requirements for the current project are so slim that's a trivial concern).

Finally, the workflow can simply specify the tools necessary to get its own test suite running, rather than have an external dependency on a requirements file to do so. The workflow knows which commands it wants to run to conduct testing, so I think the workflow needs to know how to obtain those tools in the first place, without the opportunity for someone to break it by making a change to a requirements file.

So, it ends up being a little redundant, but I think that redundancy in the case of testing helps ensure the CI workflow is a bit more resilient, to the point that the CI is almost its own application with zero knowledge of how `mws` operates.